galleries/examples/lines_bars_and_markers/timeline.py

```diff
@@ -64,32 +66,43 @@
 #
 # Note that Matplotlib will automatically plot datetime inputs.
 
-
-# Choose some nice levels
-levels = np.tile([-5, 5, -3, 3, -1, 1],
-                 int(np.ceil(len(dates)/6)))[:len(dates)]
-
-# Create figure and plot a stem plot with the date
+# Choose some nice levels: alternate minor releases between top and bottom, and
+# progressievly shorten the stems for bugfix releases.
+levels = []
+major_minor_releases = sorted({name[:4] for name in names})
+for name in names:
+    major_minor = name[:4]
+    bugfix = int(name[5])
+    h = 1 + 0.8 * (5 - bugfix)
+    level = h if major_minor_releases.index(major_minor) % 2 == 0 else -h
+    levels.append(level)
+
+# The figure and the axes.
 fig, ax = plt.subplots(figsize=(8.8, 4), layout="constrained")
 ax.set(title="Matplotlib release dates")
 
-ax.vlines(dates, 0, levels, color="tab:red")  # The vertical stems.
-ax.plot(dates, np.zeros_like(dates), "-o",
-        color="k", markerfacecolor="w")  # Baseline and markers on it.
-
-# annotate lines
+# The vertical stems.
+ax.vlines(dates, 0, levels,
+          color=[("tab:red", 1 if name.endswith(".0") else .5) for name in names])
+# The baseline.
+ax.axhline(0, c="k")
```

Comment:
Take or leave - I slightly perfer not to use shortcuts in examples to improve readability.
```suggestion
ax.axhline(0, color="black")
```